node/core/pvf/src/execute/worker.rs

```diff
@@ -142,11 +142,8 @@ pub async fn start_work(
 				target: LOG_TARGET,
 				worker_pid = %pid,
 				validation_code_hash = ?artifact.id.code_hash,
-				"execution worker exceeded allotted time for execution",
+				"execution worker exceeded lenient timeout for execution, child worker likely stalled",
 			);
-			// TODO: This case is not really a hard timeout as the timeout here in the host is
-			// lenient. Should fix this as part of
-			// https://github.com/paritytech/polkadot/issues/3754.
 			Response::TimedOut
```

Comment:
Just leaving this error as-is. If we hit this case we should just treat it the same as any other timeout, i.e. kill the worker. The log above should be good enough to inform users what happened.

I had some thoughts at the time about refactoring the execution errors, like we did for preparation. However, looking at the errors here, there don't seem to be any easy wins for simplicity or semantic correctness.